GTBitsOfGood/southface
# hc/63-addLandingPageAndPrimaryCategories

## PR Add Landing Page + Primary Categories Page

Issue Number(s): #63 #64 

What does this PR change and why?

Adds a landing page that has 3 building types that are clickable and take you to another new page that has the primary categories associated with that building type. 

Clicking one of those primary categories should take you to a library page with only the cards relevant to that category.
URL navigation is dynamic and restructured to `/library/buildingType/primaryCategory`.

Card schema is updated to have buildingType and primaryCategory properties of type `string`.

### Checklist

- [x] Created a schema called BuildingType with the following properties:
        - Name
        - Array of PrimaryCategories
- [x] Created a schema called PrimaryCategory with the following properties:
        - Name
        - Array of Cards
 - [x] Created buildingType and primaryCategories field on Card schema
 - [x] In each of those Building Types, created an array of eight PrimaryCategories:
        - Site Planning (SP)
        - Resource Efficiency (RE)
        - Durability and Moisture Management (DU)
        - High-Performance Building Envelope (BE)
        - Energy Efficient HVAC Systems (ES)
        - Indoor Air Quality (IAQ)
        - Plumbing and Irrigation (PI)
        - Efficient Lighting and Appliances (LA)
        - Education and Operations (EO)
  - [x] Displays accurate cards for each building type + primary categories (see other notes)

### Initial Landing Page (/)
- [x] Three building types (multifamily, singleFamily, commercial) displayed + respective logos
- [x] Created a general building type card component which is used to render the three categories
- [x] Landing page is shown in the root URL (/)
- [x] OnClick of a building type redirects to the primary categoires page

### Primary Categories Page (/categories)
- [x] Nine primary categories displayed 
        - Site Planning (SP)
        - Resource Efficiency (RE)
        - Durability and Moisture Management (DU)
        - High Performance Building Envelope (BE)
        - Energy Efficient HVAC Systems (ES)
        - Indoor Air Quality (IAQ)
        - Plumbing and Irrigation (PI)
        - Efficient Lighting and Appliances (LA)
        - Education and Operations (EO)
 - [x] General primary category component created to render the eight  primary categories
 - [x] Primary category page is shown in the /categories URL
 - [x] OnClick of a primary category redirects to the library page
 - [x] Bread crumbs follow figma logic

### How to Test
- Navigate to `/library`
- Click on a building type
- Check that it moves to `/library/buildingType` with the right categories ascribed to that building type
- Click on a category
- Check that it moves to `/library/buildingType/categoryInitials` with the right cards ascribed to that category
- Check that breadcrumbs lead to the right destinations and show up with the right values.

## Other notes
I edited the next.config.js to remove the automatic redirect from `/`, but it had weird caching issues and took a lot of cache clearing and rebuilding to fix. If it won't let you go to the page, try clearing the cache and rebuilding the prod version.
Sometimes there are also problems with loading the /library URL since it used to be something else. 

Constants file created for storing conversion between initials and category names as well as the internal names of buildingTypes versus the displayed names. Any modification of these will have to be done in the Card model as well.

Using getInitialProps instead of getServerSideProps because it allows for more raw request content.

@tawsifkamal should primaryCategories be an array on the Card schema instead of a string so you can have multiple categories for a single card? Same question for buildingType.
